scripts/pre_push_hook.py

```diff
@@ -373,12 +392,22 @@ def main(args=None):
                         'Push failed, please correct the linting issues above.')
                     sys.exit(1)
             frontend_status = 0
+            travis_ci_check_status = 0
             if does_diff_include_js_or_ts_files(files_to_lint):
                 frontend_status = start_python_script(FRONTEND_TEST_SCRIPT)
             if frontend_status != 0:
                 python_utils.PRINT(
                     'Push aborted due to failing frontend tests.')
                 sys.exit(1)
+            if does_diff_include_travis_yaml_or_protractor_conf_file(
```

Comment:
> We should write a check that detects cases where the list of e2e tests in .travis.yml does not match the files that exist in core/tests/protractor.

According to the expected behavior /expectations stated in the issue^ we should validate all the files exist in the core/tests/protractor should be mentioned in the travis.yaml file. Are we going to enforce that all the existing e2e test files should always be in use? Assume someone deletes a test suite from protractor.conf.js and .travis.yaml but the e2e test file still exists (or vise-versa i.e, someone added new test file but that isn't in yaml or protractor config file.).

One more point:
Assume someone only changes the protractor.conf.js file:
Before:
```
oneword: [
    'protractor/oneword.js'
  ],
```
After:
```
oneword: [],
```

In this case we will still have the oneword.js file and the oneword suite but it's not running, will this get caught via checks?

Reply:
I have added one more check, it will cover these cases that you mentioned. check_e2e_tests_are_covered_in_ci, line 165-175

Reply:
Thanks for adding that test, I've a few follow-up questions:
1. This test will only be triggered when there is a change in .travis.yml and protractor.config.js file, so what if the user renames the protractor filename, will this be get caught?

2. Is the TRAVIS_CI_PROTRACTOR_CHECK_SCRIPT runs on travis as well? or it just runs locally?

Reply:
@DubeySandeep As for point no.1, No, it will not be caught.
2. It just runs locally

Reply:
For point 2: Only running this locally will not be a good idea considering there is a chance (I've seen this before) that the pre_commit hooks didn't work locally and if this is missed once it won't be caught until someone again changes .travis.yml file.

For point 1: I would suggest, you can do either:
1. Only run this locally if the .travis.yml and protractor.config.js file change and run this check for every PR on travis.  
2. Locally run this check if any files in test/protractor changes (or any js file changes).

Reply:
@DubeySandeep Regaring point 1, I have changed the test to run if some change happen to a javascript file so point 1 is covered.
Regarding point 2, I agree a lot of times the hooks do not trigger, so we can run this on Travis or maybe CircleCI. @kevinlee12 Hey Kevin, what do you suggest?

Reply:
Another option is to run this on Github Actions. We have the linter there at the moment, it’s proven to be fairly reliable at running.

Reply:
@kevinlee12 What changes should I make in order for this test to run on Github Actions?